themes/gatsby-theme-blog/gatsby-node.js

```diff
@@ -1,116 +1,158 @@
 const fs = require(`fs`)
-const Promise = require(`bluebird`)
-const _ = require(`lodash`)
-const { createFilePath } = require(`gatsby-source-filesystem`)
+const path = require(`path`)
+const mkdirp = require(`mkdirp`)
+const Debug = require(`debug`)
 
-exports.onPreBootstrap = ({ reporter }) => {
-  const dirs = [`content`, `content/posts`, `content/assets`]
+// @TODO document theme options:
+// - takes an option `contentDir` for where the post content will live. Defaults to `posts`
+// - takes an option `assetDir` for where required assets will live. Defaults to `assets`
+
+const debug = Debug(`gatsby-theme-blog`)
+
+// These are customizable theme options we only need to check once
+let basePath
+let contentDir
+let assetDir
+
+// These templates are simply data-fetching wrappers that import components
+const PostTemplate = require.resolve(`./src/templates/post`)
+const PostsTemplate = require.resolve(`./src/templates/posts`)
+
+// Ensure that content directories exist at site-level
+exports.onPreBootstrap = ({ store }, themeOptions) => {
+  const { program } = store.getState()
+
+  basePath = themeOptions.basePath || `/`
+  contentDir = themeOptions.contentDir || `posts`
```

Comment:
Didn't we choose `contentPath` for this? That's what I'm teaching in the egghead course. 😕

Reply:
You are right, I double-checked - That's my mistake. Will change.